src/components/modals/WcSmartContractModal.tsx

```diff
@@ -202,3 +194,37 @@ async function wcRequestResponse(wallet: EdgeCurrencyWallet, uri: string, approv
     throw e
   }
 }
+
+const asWcRpcPayload = asObject({
+  id: asEither(asString, asNumber),
+  method: asValue('personal_sign', 'eth_sign', 'eth_signTypedData', 'eth_sendTransaction', 'eth_signTransaction', 'eth_sendRawTransaction'),
```

Comment:
TypeScript is probably going to infer `string` for this parameter. If you want it to infer the proper enum type, put `const asWcRpcPayloadMethod = asValue(...)` on its own line, and then use the named `asWcRpcPayloadMethod` here inside the object.

For some reason, putting `asValue` on its own line helps TypeScript do better inference.